File: app/src/main/java/com/example/simulador_dio/ui/MainActivity.java
package com.example.simulador_dio.ui;

import android.os.Bundle;
import android.util.Log;


import androidx.appcompat.app.AppCompatActivity;

import com.example.simulador_dio.R;
import com.example.simulador_dio.data.MatchesApi;
import com.example.simulador_dio.databinding.ActivityMainBinding;
import com.example.simulador_dio.domain.Match;
import com.google.android.material.snackbar.Snackbar;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private MatchesApi matchesApi;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setupHttpClient();
        setupMatchesList();
        setupMatchesRefresh();
        setupFloatActionButton();
    }

    private void setupHttpClient() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://otaviogo.github.io/matches-simulator-api/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        matchesApi = retrofit.create(MatchesApi.class);
    }

    private void setupMatchesList() {
        matchesApi.getMatches().enqueue(new Callback<List<Match>>() {
            @Override
            public void onResponse(Call<List<Match>> call, Response<List<Match>> response) {
                if (response.isSuccessful()){
                    List<Match> matches = response.body();
                    Log.i("SIMULATOR","Deu tudo certo! partidas = " + matches.size());
                } else{
                    showErrorMessage();
                }
            }

            @Override
            public void onFailure(Call<List<Match>> call, Throwable t) {
                showErrorMessage();
            }
        });
    }

    private void setupMatchesRefresh() {
        //TODO Atualizar as partidas na ação de swipe
    }

    private void setupFloatActionButton() {
        //TODO Criar evento de click e simulação de partidas
    }

    private void showErrorMessage() {
        Snackbar.make(binding.fabSimulate, R.string.error_api, Snackbar.LENGTH_LONG).show();
    }
}

File: app/src/main/java/com/example/simulador_dio/ui/MainActivity.kt
package com.example.simulador_dio.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.simulador_dio.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupMatchesList()
        setupMatchesRefresh()
        setupFloatingActionButton()
    }

    private fun setupMatchesList() {
        TODO("Listar as partidas consumindo nossa API>")
    }

    private fun setupMatchesRefresh() {
        TODO("Atualizar as partidas na ação de swipe")
    }

    private fun setupFloatingActionButton() {
        TODO("Criar evento de click e simulação de partidas")
    }
}